tools/DB_date_zeros_removal.php

```diff
@@ -90,10 +90,10 @@
 
     if ($field['COLUMN_DEFAULT']=='0000-00-00') {
         echo "The script will modify the date schema for TABLE: `".$field['TABLE_NAME']."` FIELD: `".$field['COLUMN_NAME']."` to default to NULL\n";
-        $alters .= "ALTER TABLE ".$db->escape($field['TABLE_NAME'])." MODIFY ".$db->escape($field['COLUMN_NAME'])." ".$field['COLUMN_TYPE']." DEFAULT NULL;\n";
+        $alters .= "ALTER TABLE ".$db->escape($field['TABLE_NAME'])." MODIFY ".$db->escape($field['COLUMN_NAME'])." ".$field['COLUMN_TYPE']." NULL DEFAULT NULL;\n";
```

Comment:
I'm a little surprised this line works because w3schools documentation says the mySQL syntax is "ALTER TABLE table_name
MODIFY **_COLUMN_** column_name datatype;". The COLUMN here is missing

Reply:
looking at the rest of LORIS, there's a bit of both with "COLUMN" and without it. should we keep consistent with the documentation?

Reply:
I would not change that as part of the bugfix just in case theres different behaviours

What does mysql say ?

Reply:
seems the "COLUMN" is optional

Reply:
Voila

Reply:
Could also collapse the if and elseif in on if (x && y) as both option do exactly the same thing

Reply:
yes, I'm making an other version of this tool soon since this one only checks for  zero dates. the next one will also check for zero IN dates as per MySQL's "strict" mode